### PR TITLE
fix: Bump failure counter on job error

### PR DESCRIPTION
**Pull request by tomasfarias**
Also bump `webhook_jobs_failed` on job error. Maybe we should have a separate counter for this? `PgJobError` usually indicates something on our end, so it could be worth it to have a distinction.

EDIT: I've done that, there is now a `webhook_jobs_database_error` counter.

### CODE REVIEW

File: hook-worker/src/worker.rs
Comment (bretthoerner) [inline hook-worker/src/worker.rs]: It's minor, but I think we should have a different enum entry like `DatabaseError` too. Just helps for grep in the future.

**Review by bretthoerner — APPROVED** [review]

File: hook-worker/src/worker.rs
Comment (bretthoerner) [reply hook-worker/src/worker.rs]: Doesn't have to be for this PR btw.

edit: It might help for distinguishing PgJobError or DatabaseError in tests, i.e. here: https://github.com/PostHog/rusty-hook/pull/36/files#diff-3c41c606a4d1c83ee33ba25992dd4d7a4288bd20270e68f210e086a1101462acR335

Comment (bretthoerner) [reply hook-worker/src/worker.rs]: Hmm, I was thinking in my head things like `complete()` returned a `sqlx::Error` but I guess those are wrapped internally:

https://github.com/PostHog/rusty-hook/blob/ead1bf28ef8d092848cceedf8e2ae8867b02276d/hook-common/src/pgqueue.rs#L30-L37

It's a little weird that `RetryInvalidError` will get treated as a DB error here. Although I guess that should never happen, since we're just calling `complete`/`fail`, it feels like the types could help us better here somehow.

Comment (tomasfarias) [reply hook-worker/src/worker.rs]: Yeah, I agree we should differentiate between `RetryInvalidError` and database errors. `RetryInvalidError` is the only special error as it returns the job itself to be failed (otherwise we would be consuming it), so I'd split it off this enum and rename the enum to `DatabaseError` or something similar.

Comment (tomasfarias) [reply hook-worker/src/worker.rs]: Happy to update this PR, will need a bit of time as I have some support tickets to work through.

Comment (bretthoerner) [reply hook-worker/src/worker.rs]: > Happy to update this PR, will need a bit of time as I have some support tickets to work through.

No big deal, it was mostly just a thought. Feel free to merge this if you want too, either way.

**Comment by bretthoerner** [comment]
I totally forgot about this one, @tomasfarias did you want to merge this as-is?

**Comment by tomasfarias** [comment]
@bretthoerner

Not sure if tests will pass, but given our conversation I've refactored `PgQueueError` to separate `RetryInvalidError` into its own struct (at the same time making type signatures simpler for methods that aren't `retry`). Now it should be easier to differentiate between `DatabaseError`s and `RetryInvalidError`.

**Comment by tomasfarias** [comment]
Looks like test failures are (for now) due to me making `new_from_pool` non-fallible (as it could not fail). Can clean those up later...

**Review by bretthoerner — APPROVED** [review]

**Comment by tomasfarias** [comment]
Waiting for #60 to be merged as there will be conflicts to fix, and makes more sense to tackle them here.

**Comment by tomasfarias** [comment]
Rebased on main after #60 was merged. This should be fine to merge too assuming tests are green!

**Pull request merged**
